### ISSUE TITLE
  Array#sort_custom should return an error instead of failing silently

### ISSUE THREAD

**Issue by ra314** [godotengine/godot-proposals#9568]
### Describe the project you are working on

GDScript

### Describe the problem or limitation you are having in your project

I used array.sort_custom, the type hints on the callable were incorrect and the function failed silently. This made it very hard to debug why my game wasn't behaving as expected.

var players : Array[String] = ["player1", "player2"]
players.sort_custom(func(o1: int, o2: int): return o1 > o2 ) 
Ought to error because of the mismatch types

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

array.sort_custom should return err.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

There are many existing functions in godot that return error code, this could behave similar to them.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I think the return value if added to this function, can be ignored by existing calls to it.

### Is there a reason why this should be core and not an add-on in the asset library?

This is a core function.

**Comment by KoBeWi** [comment]
It does fail with an error:
![image](https://github.com/godotengine/godot-proposals/assets/2223172/0a6e878b-906f-484e-aacb-9fa7884441f4)

**Comment by ra314** [comment]
Right, this is how me and my friend noticed that this was the problem. But ideally if it had a return value, I would've had an assert on that like I do with all other functions which return an error value. Additionally the logs in the debugger don't show which line of code is causing the error, so it's not particularly useful.
